[홍현진] Mission 6: 로그인 JWT 발급

hjz1010 opened the pull request:
>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
 - LoginView에 로그인 JWT 발급 단계 추가
		 
## 수정 사항들 자세한 내용
- bcrypt.checkpw()메소드로 비밀번호 확인 로직 변경
- 로그인 패스워드 확인 및 JWT 발급 단계에서 user 정보가 반복 사용되고 있어서
   get() 메소드를 사용하여 user 변수 할당하고 사용하는 방식으로 변경
- 로그인 성공 시 JWT payload에 User id 를 담아서 발급
- 발급된 JWT를 응답 데이터에 반환
    
## 기타 질문 및 특이 사항

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
조건을 만족한다면 각 빈칸에 x 표시 해주세요
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

hjz1010 committed:
로그인 JWT 발급 1단계

- bcrypt.checkpw()메소드로 비밀번호 확인 로직 변경

hjz1010 committed:
로그인 JWT 발급 2단계 +a

- 로그인 성공 시 JWT payload에 User id 를 담아서 발급
- 발급된 JWT를 응답 데이터에 반환
+ 로그인 패스워드 확인 및 JWT 발급 단계에서 user 정보가 반복 사용되어, get() 메소드를 사용하여 user 변수 할당하고 사용하는 방식으로 변경

hjz1010 committed:
오타 수정 (endcode 너 왜 뭔데 자꾸 나와)

hjz1010 committed:
merge MISSION 5

Merge branch 'main' into feature/hyeonjin-auth2

j-eum commented:
LGTM 🚀

Jon9Beom commented on students/hyeonjin/users/views.py:
```diff
@@ -1,11 +1,10 @@
 from django.views import View
 
 from users.models import User
+from my_settings import SECRET_KEY, ALGORITHM
```
```suggestion
from django.conf import settings
```
현진님~ `HS256`, `SECRET_KEY`와 같은 환경변수를 `my_settings.py` 파일로 관리하고 있습니다. `my_settings.py` 파일은 환경변수 관리로서의 역할만 하는 파일로서 `views.py` 또는 `models.py`에서 직접 import해서 사용하시면 `views.py` 파일이 `my_settings.py` 파일에 의존성이 생기게 됩니다.

만약에 환경변수 관리를 my_settings.py가 아니라 아래와 같이 다른 방법으로 관리하게 되면 views.py에 코드도 전부 수정해야되는 것이죠.

```python
# settings.py

DATABASES = {
    ..
    "PASSWORD" : os.environ.get("DB_PASSWORD")
}
```
그러므로 `my_settings.py`에서 환경변수를 import해서 사용하실 때에는 지금처럼 `from my_settings import ALGORITHM, SECRET_KEY` 방법이 아니라 `from django.conf import settings`를 사용해서 환경변수를 불러와서 사용해주세요.

hjz1010 replied:
자세한 설명 감사합니다! 🙂👍

Jon9Beom commented on students/hyeonjin/users/views.py:
```diff
@@ -56,11 +55,16 @@ def post(self, request):
                 return JsonResponse({"message":"INVALID_USER"}, status=401)
-            return JsonResponse({"message":"SUCCESS"}, status=200)    
 
+            encoded_jwt = jwt.encode({'user_id': user.id}, SECRET_KEY, ALGORITHM)
```
```suggestion
            encoded_jwt = jwt.encode({'user_id': user.id}, settings.SECRET_KEY, settings.ALGORITHM)
```
위의 리뷰 내용에 따라 다음과 같이 수정 할 수 있습니다.

hjz1010 committed:
수정요청 반영
- 환경변수를 my_settings.py에서 직접 import하면 의존성이 생기므로
- settings.py에 import 한 뒤
- views.py에서 사용할 때는  from django.conf import settings 방식으로 불러와서 사용한다.

Jon9Beom commented on students/hyeonjin/users/views.py:
```diff
@@ -1,11 +1,10 @@
-import re
-import bcrypt
+import json, re, bcrypt, jwt
 
 from django.http  import JsonResponse
 from django.views import View
 
 from users.models import User
+from django.conf import settings
 
```
```suggestion

from django.http  import JsonResponse
from django.views import View
from django.conf  import settings

from users.models import User

```

현진님 모듈 import 순서 및 컨벤션 수정해주세요 🙂

hjz1010 replied:
앗! 맘이 급했습니다! 바로 수정하겠습니다:)

hjz1010 committed:
컨벤션 수정

Jon9Beom commented:
LGTM 🚀
